### PR TITLE
Add all available types of BackupTarget to run in one test

### PR DESCRIPTION
**Pull request by JacieChao**

### CODE REVIEW

**Comment by JacieChao** [comment]
WIP

Tested with kubernetes 1.8.10-gke.0 and 1.10.2-gke.3 on GKE. 
Here's the test results

+ flake8 .
+ py.test -v .
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-2.9.2, py-1.5.3, pluggy-0.3.1 -- /usr/local/bin/python
cachedir: .cache
rootdir: /integration/tests, inifile:
collecting ... collected 17 items
test_basic.py::test_hosts_and_settings PASSED
test_basic.py::test_volume_basic PASSED
test_basic.py::test_volume_iscsi_basic PASSED
test_basic.py::test_snapshot PASSED
test_basic.py::test_backup PASSED
test_basic.py::test_volume_multinode PASSED
test_csi.py::test_csi_volume_mount SKIPPED
test_csi.py::test_csi_volume_io SKIPPED
test_driver.py::test_volume_mount PASSED
test_driver.py::test_volume_io PASSED
test_engine_upgrade.py::test_engine_image PASSED
test_engine_upgrade.py::test_engine_offline_upgrade PASSED
test_engine_upgrade.py::test_engine_live_upgrade PASSED
test_engine_upgrade.py::test_engine_image_incompatible PASSED
test_ha.py::test_ha_simple_recovery PASSED
test_ha.py::test_ha_salvage PASSED
test_recurring_job.py::test_recurring_job SKIPPED
=================== 14 passed, 3 skipped in 1246.62 seconds ====================

I will test with local kubernetes server( v1.8.10 and v1.10.3 ) tomorrow.

File: manager/integration/tests/test_basic.py [hunk 1]
```diff
@@ -321,85 +321,94 @@ def test_backup(clients, volume_name):  # NOQA
...
-        credential = client.by_id_setting("backupTargetCredentialSecret")
-        credential = client.update(credential,
-                                   value=common.get_backupstore_credential())
-        assert credential["value"] == common.get_backupstore_credential()
-
-    volume.snapshotCreate()
-    snap2 = volume.snapshotCreate()
-    volume.snapshotCreate()
+    # test backupTarget for multiple settings
+    backupstores = common.get_backupstore_url()
```
Comment (yasker) [inline manager/integration/tests/test_basic.py]: Move the setting of `backupTarget` and credential to `test_backup` before calling `backup_test` should make it easier and we won't need so many indent changes (and the change of logic will be hard to spot in the diff).

Btw, can rename `setting` to e.g. `backuptarget_setting`.

File: manager/integration/deploy/backupstore/nfs-backupstore.yaml [hunk 2]
```diff
@@ -16,10 +19,14 @@ spec:
       value: /opt/backupstore
     - name: PSEUDO_PATH
       value: /opt/backupstore
-    command: ["bash", "-c", "mkdir -p /opt/backupstore && /opt/start_nfs.sh"]
+    command: ["bash", "-c", "chmod 700 /opt/backupstore && /opt/start_nfs.sh"]
```
Comment (yasker) [inline manager/integration/deploy/backupstore/nfs-backupstore.yaml]: This solve the Ganesha crash problem>

**Comment by JacieChao** [comment]
@yasker I have tested kubernetes from 1.8.10-gke.0 to 1.10.2-gke.3 on GKE and also tested with kubernetes v1.10.3 which is setup by RKE. The test results looked nice.
Here's one of test results.
```
+ flake8 .
+ py.test -v .
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-2.9.2, py-1.5.3, pluggy-0.3.1 -- /usr/local/bin/python
cachedir: .cache
rootdir: /integration/tests, inifile:
collecting ... collected 17 items
test_basic.py::test_hosts_and_settings PASSED
test_basic.py::test_volume_basic PASSED
test_basic.py::test_volume_iscsi_basic PASSED
test_basic.py::test_snapshot PASSED
test_basic.py::test_backup PASSED
test_basic.py::test_volume_multinode PASSED
test_csi.py::test_csi_volume_mount SKIPPED
test_csi.py::test_csi_volume_io SKIPPED
test_driver.py::test_volume_mount PASSED
test_driver.py::test_volume_io PASSED
test_engine_upgrade.py::test_engine_image PASSED
test_engine_upgrade.py::test_engine_offline_upgrade PASSED
test_engine_upgrade.py::test_engine_live_upgrade PASSED
test_engine_upgrade.py::test_engine_image_incompatible PASSED
test_ha.py::test_ha_simple_recovery PASSED
test_ha.py::test_ha_salvage PASSED
test_recurring_job.py::test_recurring_job SKIPPED
=================== 14 passed, 3 skipped in 1226.91 seconds ====================
```

**Comment by yasker** [comment]
LGTM. Merged, thanks.

**Pull request merged**